lib/ansible/plugins/inventory/auto.py

```diff
@@ -39,7 +39,10 @@ def verify_file(self, path):
     def parse(self, inventory, loader, path, cache=True):
         config_data = loader.load_from_file(path, cache=False)
 
-        plugin_name = config_data.get('plugin')
+        try:
+            plugin_name = config_data.get('plugin', None)
+        except AttributeError:
+            plugin_name = None
```

Comment:
Isn't this try/except redundant with the second param passed to `get()`?

Reply:
Probably `config_data` might be something that does not have `get` and throw `AttributeError`?

Reply:
can return None, so you get an attribute error on `Nonetype does not have get`